dbt/adapters/trino/connections.py

```diff
@@ -96,43 +94,14 @@ def fetchall(self):
         return None
 
     def execute(self, sql, bindings=None):
-
-        if bindings is not None:
-            # trino doesn't actually pass bindings along so we have to do the
-            # escaping and formatting ourselves
-            bindings = tuple(self._escape_value(b) for b in bindings)
-            sql = sql % bindings
-
-        result = self._cursor.execute(sql)
+        result = self._cursor.execute(sql, params=bindings)
```

Comment:
The insert statements executed on Trino while doing `dbt seed` look now like this:

```
PREPARE st_24e2ddaa611544f3a6a459e6997a5beb FROM insert into "memory"."default"."base" ("id", "name", "some_date") values
          (?,?,?),(?,?,?),(?,?,?),(?,?,?),(?,?,?),(?,?,?),(?,?,?),(?,?,?),(?,?,?),(?,?,?)
```

Is there any limitation on Trino on how many bindings can be used in a prepared statement?

Reply:
Indeed the OOTB code uses this feature only for seeds. User code can also use this feature, but as our binding char was not defined correclty and no bugs have been raised. I doubt this is actively used.

The queries are indeed split into PREPARE and then EXECUTE (with the variables), which performs the INSERT.

Prepared statement variables are sent as part of the headers. It seems that the trino server puts a hard limit on that of 2Mb. [See relevant issue](https://github.com/trinodb/trino/issues/8797). dbt actually advises against too big CSV files. 

<img width="986" alt="image" src="https://user-images.githubusercontent.com/9265503/163974433-8ca3f7fb-ffe6-4191-add6-da2a2cfcfe98.png">

The code uses following macro to define batch size for the INSERT statements:

```
{% macro trino__get_batch_size() %}
  {{ return(1000) }}
{% endmacro %}
```
The size of headers would then expand in 1000 * num columns of the seed file. I would say the hard limits should be sufficient.

What we can do is to add some integration tests for this feature? WDYT?

Reply:
I think we're good.
We already do `dbt seed` as part of the tests.
The tests in their current format are definitely not ideal, but this is a different issue.

Reply:
I meant volume tests. ensuring the prepared statements support inserting 1000 or more records.

Reply:
> What we can do is to add some integration tests for this feature?

> The tests in their current format are definitely not ideal.

FYI, I'm working currently to move some (or even all) tests to the new adapter testing framework. Especially, I would love to get rid of jaffle_shop.